test_plans/accessor_properties.asciidoc

```diff
@@ -0,0 +1,288 @@
+
+Create sycl::accessor instance `acc` with following constructors:
+
+* `accessor(buffer<dataT, 1, AllocatorT> &bufferRef,
+           const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, 1, AllocatorT> &bufferRef,
+           handler &commandGroupHandlerRef, const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef, TagT tag,
+           const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           handler &commandGroupHandlerRef, const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           handler &commandGroupHandlerRef, TagT tag,
+           const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           range<dimensions> accessRange, const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           range<dimensions> accessRange, TagT tag,
+           const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           range<dimensions> accessRange, id<dimensions> accessOffset,
+           const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           range<dimensions> accessRange, id<dimensions> accessOffset,
+           TagT tag, const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           handler &commandGroupHandlerRef, range<dimensions> accessRange,
+           const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           handler &commandGroupHandlerRef, range<dimensions> accessRange,
+           TagT tag, const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           handler &commandGroupHandlerRef, range<dimensions> accessRange,
+           id<dimensions> accessOffset, const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+* `accessor(buffer<dataT, dimensions, AllocatorT> &bufferRef,
+           handler &commandGroupHandlerRef, range<dimensions> accessRange,
+           id<dimensions> accessOffset, TagT tag,
+           const ext::oneapi::accessor_property_list<properties...> &propList = {});`
+
+For created accessor instance check the following tests:
```

Comment:
None of the tests below are specific to compile-time constant properties.  I'm guessing this is just the list of tests we perform for regular accessors?  Again, I think it is not necessary to duplicate all this testing for accessors with compile-time constant properties.

I do think we need some testing for accessors created with each of the constructors listed above, though.  I'd recommend running a simple-form `parallel_for` kerenel that takes two accessors: one `read_only` and another `write_only`.  The kernel can simply copy data from one accessor to the other.  Your test can then check that the data was copied after waiting for the kernel to complete.

Note that you will need to take care to only use the `no_init` property on the `write_only` accessor because that property is invalid for `read_only` accessors.

Reply:
Section removed as duplicate.